src/main/webapp/app/blocks/interceptor/auth.interceptor.ts

```diff
@@ -2,18 +2,22 @@ import { Observable } from 'rxjs/Observable';
 import { RequestOptionsArgs, Response } from '@angular/http';
 import { LocalStorageService, SessionStorageService } from 'ng2-webstorage';
 import { HttpInterceptor } from 'ng-jhipster';
+import {Injector} from "@angular/core";
+import {CookieService} from "angular2-cookie/core";
+import {AUTH_TOKEN_COOKIE} from "../../shared/constants/common.constants";
 
 export class AuthInterceptor extends HttpInterceptor {
 
     constructor(
-        private localStorage: LocalStorageService,
-        private sessionStorage: SessionStorageService
+        private injector: Injector,
     ) {
         super();
     }
 
     requestIntercept(options?: RequestOptionsArgs): RequestOptionsArgs {
-        const token = this.localStorage.retrieve('authenticationToken') || this.sessionStorage.retrieve('authenticationToken');
+        // retrieve token from cookie
+        const cookieService = this.injector.get(CookieService);
+        const token : any = cookieService.getObject(AUTH_TOKEN_COOKIE);
```

Comment:
Perhaps we can take this opportunity to change this bit to use the `auth-oauth2.service`'s `getToken()` method instead.

Reply:
I will check. The reason i used CookieService is to avoid cyclic dependency, since auth-oauth2.service depends on `Http` and this is an HTTP interceptor. But i think we could still use the injector to get the instance of the service. I will have a look.